src/goto-symex/symex_clean_expr.cpp

```diff
@@ -169,6 +170,51 @@ replace_nondet(exprt &expr, symex_nondet_generatort &build_symex_nondet)
   }
 }
 
+void goto_symext::lift_let(
+  statet &state,
+  const let_exprt &let_expr,
+  assignment_typet assignment_type)
+{
+  exprt let_value = let_expr.value();
+  clean_expr(let_value, state, false);
+  let_value = state.rename(std::move(let_value), ns).get();
```

Comment:
Just as a note: some time ago @kroening proposed to make `let` expressions be a scope of their own. Thus care would need to be taken as the `let` expression could have symbols of local scope even when the name overlaps with something occurring elsewhere. Not the case right now, but needs care moving forward.